### PR TITLE
Add softmax, bce, sparse ce loss

### PR DESCRIPTION
**Pull request by workofart**
Add some loss functions:
- Binary Cross Entropy for probabilties
- Binary Cross Entropy for logits
- Sparse Cross Entropy for probabilities (multiclass classification)
- Sparse Cross Entropy for logits (multiclass classification)

Add Softmax activation function, and associated backward() (gradient computation)

## Tests
Added `test_functional.py` to cover all the functions in that file
image

### CODE REVIEW

File: test/autograd/test_nn.py [hunk 1]
```diff
@@ -70,60 +66,3 @@ def test_linear(self):
                 [1, 1],
             ],
         )
-
-
-class TestActivations(TestCase):
```
Comment (workofart) [inline test/autograd/test_nn.py:75]: Moving these test cases to dedicated `test_functional.py`
